### PR TITLE
[COH-16158] Renoir tracing and memory tracking

### PR DESCRIPTION
**Pull request by palikar**
The PR:
* Adds support for the Renoir trace events
* Adds support for the Renoir counters
* Enhances that inspector so that it can do basic filtering in the chart counters panel

### CODE REVIEW

File: front_end/models/timeline_model/TimelineModel.ts [hunk 1]
```diff
@@ -1737,6 +1737,26 @@ export enum RecordType {
...
+  Coherent_ProcessLayer = 'Coherent_ProcessLayer',
+  Coherent_BatchCommands = 'Coherent_BatchCommands',
+  Coherent_DistributeLayers = 'Coherent_DistributeLayers',
+
+  Coherent_SubmitGlyphs = 'Coherent_SubmitGlyphs',
+  Coherent_SubmitManagerChanges = 'Coherent_SubmitManagerChanges',
+  Coherent_ProcessFrontendCommandsOnly = 'Coherent_ProcessFrontendCommandsOnly',
+  Coherent_DrawSDFGlyphs = 'Coherent_DrawSDFGlyphs',
+  Coherent_DrawSubLayerWithShaderFilter = 'Coherent_DrawSubLayerWithShaderFilter',
+  Coherent_Backend_Execute = 'Coherent_Backend_Execute',
```
Comment (gashtio) [inline front_end/models/timeline_model/TimelineModel.ts]: Not a big deal, but every other event **doesn't** have an underscore in the name, what's up with this one?

File: front_end/panels/timeline/CountersGraph.ts [hunk 2]
```diff
@@ -175,8 +196,7 @@ export class CountersGraph extends UI.Widget.VBox {
       for (const name in counters) {
         const counter = this._countersByName.get(name);
         if (counter) {
-          counter.appendSample(event.startTime, counters[name]);
-        }
+          counter.appendSample(event.startTime, counters[name]);}
```
Comment (gashtio) [inline front_end/panels/timeline/CountersGraph.ts]: revert?

**Review by gashtio — APPROVED** [review]
Seems legit

File: front_end/models/timeline_model/TimelineModel.ts (hunk 1 above)
Comment (palikar) [reply front_end/models/timeline_model/TimelineModel.ts]: the old tracing in Renoir was like so 
```c++
RENOIR_PROFILE_SCOPE_L2(Commands, Backend_Execute);
```
and I copied everything mechanically. I'll fix this :D

File: front_end/panels/timeline/CountersGraph.ts (hunk 2 above)
Comment (palikar) [reply front_end/panels/timeline/CountersGraph.ts]: sure, fixed

File: front_end/panels/timeline/CountersGraph.ts [hunk 3]
```diff
@@ -127,18 +131,35 @@ export class CountersGraph extends UI.Widget.VBox {
...
+    this._showSimpleCounters = Common.Settings.Settings.instance().createSetting('timelineCounters', false);
+
+    this._showMemoryCounters.addChangeListener(this._onShowCountersChanged, this);
+    this._showSimpleCounters.addChangeListener(this._onShowCountersChanged, this);
+  }
+
+  _onShowCountersChanged(): void {
+
+    for (const counterUI of this._counterUI)
+    {
```
Comment (PlamenMinev) [inline front_end/panels/timeline/CountersGraph.ts]: In JS '{' are on the same line. Can you refactor them?
This includes functions, ifs, for...

**Review by PlamenMinev — APPROVED** [review]

File: front_end/panels/timeline/CountersGraph.ts (hunk 3 above)
Comment (palikar) [reply front_end/panels/timeline/CountersGraph.ts]: sure

**Review by Tony-CL — APPROVED** [review]
Reminders:
- Task about tests?
- Something to be mentioned in the documentation?

**Pull request merged**